azure/scope/machinepool.go

```diff
@@ -688,3 +701,39 @@ func (m *MachinePoolScope) UpdatePatchStatus(condition clusterv1.ConditionType,
 		conditions.MarkFalse(m.AzureMachinePool, condition, infrav1.FailedReason, clusterv1.ConditionSeverityError, "%s failed to update. err: %s", service, err.Error())
 	}
 }
+
+// PatchCAPIMachinePoolObject persists the capi machinepool configuration and status.
+func (m *MachinePoolScope) PatchCAPIMachinePoolObject(ctx context.Context) error {
```

Comment:
just like in #2444, we need to make sure that the controller has appropriate rbac permissions to patch MachinePools

Reply:
is this something we need to do additionally? The changes to `azuremachinepool_controller.go` and `rbac.yaml`  in #2444 should be enough to work also in this case, if I'm not mistaken.